Created Server View

HenrikhB opened the pull request:
Created Server Views for handling GET,POST,PUT,DELETE.

Commit:
Created Server Views

Views for handling GET, POST, PUT, DELTE requests.

hutsulyk commented on moninag/server/models.py:
```diff
@@ -1,30 +1,156 @@
+        filters = {}
+
+        if states:
+            filters['status__in'] = states
```
Do you have `status` field? :upside_down_face:

hutsulyk commented on moninag/server/urls.py:
```diff
@@ -0,0 +1,11 @@
+urlpatterns = [
+    url(r'^$', csrf_exempt(views.ServerView.as_view()), name='server'),
+    url(r'^(?P<server_id>\d+)/$', csrf_exempt(views.ServerView.as_view()), name='server'),
+]
```
Add end line.

HenrikhB replied:
done

lhalam commented on moninag/server/models.py:
```diff
@@ -1,30 +1,156 @@
+        :return: Created server for success, None otherwise.
+        """
+
+        try:
```
Remove user search unit,
The method receives the existing user_id

lhalam commented on moninag/server/models.py:
```diff
@@ -1,30 +1,156 @@
+    state = models.CharField(max_length=20, choices=STATE_CHOICES, default='')
+    user = models.ForeignKey(CustomUser, default=1, on_delete=models.CASCADE)
+
+    @classmethod
```
use @staticmetod

lhalam commented on moninag/server/models.py:
```diff
@@ -1,30 +1,156 @@
+        return server
+
+    @classmethod
+    def update(cls, id, name, address, state, user_id):
```
Are all parameters required?

lhalam commented on moninag/server/models.py:
```diff
@@ -1,30 +1,156 @@
+        """
+
+        # Check if server exist allready
+        if cls.objects.filter(id=id).count() == 0:
```
Remove user search unit,
The method receives the existing user_id and server_id

lhalam reviewed (commented):
fix it all by analogy to comments

Commit:
Updated Server View

Commit:
Added subset validate method

validate_subdict method checks whether the dict fits the requirements set

hutsulyk commented on moninag/utils/validators.py:
```diff
@@ -10,3 +10,17 @@ def validate_dict(dictionary, requirements):
+        bool: The return value. True if valid, False otherwise.
+    """
+
+    return bool(set(dictionary.keys()).issubset(requirements))
```
`issubset()` already return bool :slightly_smiling_face:

Commit:
Updated Server Views

For now Server Views have strict dependency to user session (user_id = request.user.id)

hutsulyk commented on moninag/server/models.py:
```diff
@@ -1,30 +1,139 @@
+        :return: The return value. Updated server for success, None otherwise.
+        """
+
+        # Check if server exist allready
```
Old comment :slightly_smiling_face:

hutsulyk commented on moninag/server/models.py:
```diff
@@ -1,30 +1,139 @@
+            self.state = state
+        self.save()
+
+        return self
```
Is it necessary to return self?

hutsulyk commented on moninag/server/views.py:
```diff
@@ -1,2 +1,141 @@
+                else:
+                    return HttpResponse(code=403)
+            else:
+                HttpResponse(code=400)
```
Did you miss `return`? :slightly_smiling_face:

hutsulyk commented on moninag/server/views.py:
```diff
@@ -1,2 +1,140 @@
+            return JsonResponse(json_response, status=201)
+
+        json_response['error'] = 'Server was not created.'
+        return JsonResponse(json_response, status=400)
```
When `if server:` will be not true?

hutsulyk commented on moninag/server/views.py:
```diff
@@ -1,2 +1,140 @@
+            if server.user.id == request.user.id:
+                server.update(**server_dict)
+                json_response['response'] = server.to_dict()
+                return JsonResponse(json_response, status=200)
```
Return 403 `if server.user.id == request.user.id:` False

HenrikhB replied:
@mhumetskyi fixed

hutsulyk commented on moninag/server/views.py:
```diff
@@ -1,2 +1,140 @@
+            if count:
+                return HttpResponse(status=200)
+        else:
+            return HttpResponse(status=400)
```
status = 403 :slightly_smiling_face:

hutsulyk commented on moninag/server/views.py:
```diff
@@ -1,2 +1,142 @@
+                return JsonResponse(json_response, status=404)
+
+        # Get all servers
+        if request.user.id:
```
`if request.user.is_authenticated:` :slightly_smiling_face:

lhalam replied:
do not need

lhalam commented on moninag/server/models.py:
```diff
@@ -1,30 +1,136 @@
+    def get_by_user_id(user_id):
+        """
+        :param user_id:
+        :return: return Server
```
`:param user_id:` type param: Int or user object?
`:return: return Server` Returns only one server or server list?

lhalam commented on moninag/server/views.py:
```diff
@@ -1,2 +1,142 @@
+            json_response['error'] = 'Incorect JSON format.'
+            return JsonResponse(json_response, status=400)
+
+        if request.user.id:
```
No need to check
Update implementing
